File: Sorting& Searching/BinarySearch.py
def BinarySearch(nums, target):
    left = 0
    right = len(nums) - 1
    mid = (left + right) // 2
    while (left <= right):
        if (nums[mid] == target):
            return mid
        elif (nums[mid] > target):
            right = mid - 1
        else:
            left = mid + 1
        mid = (left + right) // 2
    return -1

nums = [1, 5, 9, 54, 88, 99, 888]
target = 0
print(BinarySearch(nums, target))

File: Sorting& Searching/BinarySearch.cpp
# include <iostream>
# include <vector>
using namespace std;

int BinarySearch(vector<int> nums, int target){
    int left = 0;
    int right = nums.size() - 1;
    int mid = (left + right) / 2;
    while (left <= right){
        if (nums[mid] == target)
            return mid;
        else if (nums[mid] > target)
            right = mid - 1;
        else
            left = mid + 1;
        mid = (left + right) / 2;
    }
    return -1;
}

int main(){
    vector<int> number = {0, 22, 55, 99, 777, 999, 10000};
    int target;
    cin >> target;
    cout << BinarySearch(number, target);
    return 0;
}